README.md

```diff
@@ -4,6 +4,19 @@
 
 **Babelfish is a self-hosted server for source code parsing.** Babelfish can parse any file, in any language, extract an ASTs (Abstract Syntax Tree) from it, and convert it to a [**UAST**](./uast/specification.md) (Universal Abstract Syntax Tree).
 
+## Motivation and Scope
+
+Babelfish was born as a solution for massive code analysis. The goal: analyzing
+all source code from every repository, for every revision.
+
+Current **scope is parsing of single files in any programming language**
+that produces a universal abstract syntax tree. This helps us to run it at scale.
+
+This scope might expand in the future to full project analysis, where source code
+can be analyzed with its full context, an not just per-file. However, this would
```

Comment:
I find this last sentence a bit overcomplicated. Maybe something along _Once Babelfish with its current scope is mature enough, we may consider starting such effort_.